apps/src/sites/code.org/pages/public/teacher-dashboard/index.js

```diff
@@ -224,7 +225,10 @@ function main() {
 
     $scope.sections = sectionsService.query();
 
-    $scope.sections.$promise.then(function ( sections ){
+    $scope.sections.$promise.then(sections => {
+      $scope.sections.forEach(section => {
+        section.assign_id = $scope.getAssignmentId(section);
+      });
```

Comment:
assign_id is the approach i've taken for dealing with the fact that scripts/courses each have their own id space.